django/test/testcases.py

```diff
@@ -274,6 +284,17 @@ def __call__(self, result=None):
         """
         self._setup_and_call(result)
 
+    def __getstate__(self):
+        """
+        Make SimpleTestCase pickable for parallel tests using subtests.
+        """
+        attributes = self.__dict__.copy()
+        # _outcome and _subtest cannot be tested on picklebility, since they contain
+        # the TestCase, and therefore itself, leading to an infinite recursion.
+        attributes["_outcome"] = None
+        attributes["_subtest"] = None
+        return {k: v for k, v in attributes.items() if is_pickable(v)}
```

Comment:
Maybe we should do this only when `_outcome` is set, to [inject directly into gathering test results](https://github.com/python/cpython/blob/1aec0644447e69e981d582449849761b23702ec8/Lib/unittest/case.py#L628-L659) :thinking: 
draft:
```diff
diff --git a/django/test/testcases.py b/django/test/testcases.py
index c7ecf166b5..4b29ff909c 100644
--- a/django/test/testcases.py
+++ b/django/test/testcases.py
@@ -318,9 +318,13 @@ class SimpleTestCase(unittest.TestCase):
         attributes = self.__dict__.copy()
         # _outcome and _subtest cannot be tested on picklebility, since they contain
         # the TestCase, and therefore itself, leading to an infinite recursion.
-        attributes["_outcome"] = None
-        attributes["_subtest"] = None
-        return {k: v for k, v in attributes.items() if is_pickable(v)}
+        outcome = attributes["_outcome"] is not None
+        if outcome:
+            attributes["_outcome"] = None
+            attributes["_subtest"] = None
+            return {k: v for k, v in attributes.items() if is_pickable(v)}
+        else:
+            return attributes
 
     def debug(self):
         """Perform the same as __call__(), without catching the exception."""
```